compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -411,6 +411,20 @@ impl<'a> Parser<'a> {
     }
 
     pub(super) fn expected_ident_found(&self) -> DiagnosticBuilder<'a, ErrorGuaranteed> {
+        if self.prev_token.is_keyword(kw::Enum) && self.token.is_keyword(kw::Struct) {
```

Comment:
Consider moving this to `parse_item_enum`, since it's a bit specific to `enum`s to be located in `expected_ident_found` I think.

Reply:
Much better i think